jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectGeneratorIntelliJShims.kt

```diff
@@ -43,37 +57,108 @@ class SamProjectBuilder(private val generator: SamProjectGenerator) : ModuleBuil
     override fun setupRootModel(rootModel: ModifiableRootModel) {
         val settings = generator.peer.settings
 
-        settings.template.setupSdk(rootModel, settings)
-
         // Set module type
         val selectedRuntime = settings.runtime
         val moduleType = selectedRuntime.runtimeGroup?.getModuleType() ?: ModuleType.EMPTY
         rootModel.module.setModuleType(moduleType.id)
 
-        val contentEntry: ContentEntry = doAddContentEntry(rootModel) ?: throw Exception(message("sam.init.error.no.project.basepath"))
-        val outputDir: VirtualFile = contentEntry.file ?: throw Exception(message("sam.init.error.no.virtual.file"))
+        val contentEntry = doAddContentEntry(rootModel) ?: throw Exception(message("sam.init.error.no.project.basepath"))
+        val outputDir = contentEntry.file ?: throw Exception(message("sam.init.error.no.virtual.file"))
 
-        // ModifiableRootModel takes a final ProjectRootManagerImpl which has a final project, so we have guaranteed access to project here
+        val project = rootModel.project
         ProgressManager.getInstance().run(
-            object : Task.Backgroundable(rootModel.project, message("sam.init.generating.template"), false) {
+            object : Task.Backgroundable(project, message("sam.init.generating.template"), false) {
                 override fun run(indicator: ProgressIndicator) {
                     ModuleRootModificationUtil.updateModel(rootModel.module) { model ->
                         val samTemplate = settings.template
-                        samTemplate.build(project, rootModel.module.name, selectedRuntime, settings.schemaParameters, outputDir)
-                        VfsUtil.markDirtyAndRefresh(false, true, true, outputDir)
-                        runInEdt {
-                            try {
-                                samTemplate.postCreationAction(settings, outputDir, model, generator.defaultSourceCreatingProject, indicator)
-                            } catch (t: Throwable) {
-                                LOG.error(t) { "Exception thrown during postCreationAction!" }
-                            }
-                        }
+
+                        runSamInit(project, rootModel.module.name, samTemplate, selectedRuntime, generator.schemaPanel.schemaInfo(), outputDir)
```

Comment:
did we already fix the issue where a new project looks empty until indexing happens? else should investigate at some point

Reply:
That is fixed

Reply:
I fixed that a few weeks ago :D